firedrake/checkpointing.py

```diff
@@ -708,7 +708,14 @@ def save_function(self, f, idx=None):
             mode (non-timestepping); for each function of interest,
             this method must always be called with the idx parameter
             set or never be called with the idx parameter set.
+        :arg name: optional alternative name to save the function under
         """
+        if name:
+            g = Function(f.function_space(), name=name)
+            g.assign(f)
+            self.save_function(g, idx=None)
+            return
```

Comment:
Rather than copying the data, you can make a Function which shares the same data space but has a different name. I also think you should be passing `idx` through rather than forcing it to None. i.e.
```
if name:
    return self.save_function(Function(f.function_space(), val=f, name=name)
```
I'm returning the result even though it's `None` just so that this will do the right thing if someone in the future decides that `save_function` should return something.

Reply:
You're definitely right about idx (silly mistake on part) and the return value. But with your proposed change, the test fails -- I find that `Function(f.function_space(), val=f, name=name).name()` is equal to `f.name()` rather than the supplied `name`.

Reply:
Alright I figured it out, you need to do `val=f.dat` and then it's fine.